### PR TITLE
[WIP] ALLUXIO-1596 Add test coverage for NettyRemoteBlockWriter 

### PR DESCRIPTION
**Pull request by caadecarvalho**
Creation of unit test for NettyRemoteBlockWriterTest.java

### CODE REVIEW

**Comment by caadecarvalho** [comment]
Jenkins, test this please.

**Comment by aaudiber** [comment]
Hey @caadecarvalho, it looks like something happened to the file to mess with the spacing and turn all the words to uppercased - could you fix up the file so it's easier to read?

**Comment by caadecarvalho** [comment]
Hi @aaudiber , I took a look on the file and it is OK as bellow:
image

File: core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+   */
+  @Before
+  public void before() throws IOException {
+    mInetSocketAddress = ClientContext.getMasterAddress();
+    mNettyRemoteBlockWriter = PowerMockito.spy(new NettyRemoteBlockWriter());
+    mNettyRemoteBlockWriter.open(mInetSocketAddress, BLOCK_ID, SESSION_ID);
+  }
+
+  /**
+   * Sets a rule for {@link IOException} exception.
```
Comment (aaudiber) [inline core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java]: `The exception expected to be thrown` (no need to constrain it to `IOException` only here)

File: core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+   */
+  @Rule public ExpectedException mThrowsMessage = ExpectedException.none();
```
Comment (aaudiber) [inline core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java]: Usually we put `@Rule` on its own line, see other tests which use ExpectedException like LoginUserTest for example.

File: core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+   *
+   * @throws IOException if {@link NettyRemoteBlockWriter#write()} method exception message
+   * starts with "java.net.ConnectException: Connection refused"
+   */
+  @Test()
+  public void writeExceptionTest() throws IOException {
+    mThrowsMessage.expect(IOException.class);
+    mThrowsMessage.expectMessage(startsWith("java.net.ConnectException: Connection refused"));
+    mNettyRemoteBlockWriter.write(BYTES, OFFSET, LENGTH);
+    Assert.assertTrue(true);
```
Comment (aaudiber) [inline core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java]: This line doesn't do anything

File: core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  }
+
+  /**
+   * Tests the {@link NettyRemoteBlockWriter#write()} method.
+   *
+   * @throws IOException if {@link NettyRemoteBlockWriter#write()} method
+   */
+  @Test()
+  public void writeTest() throws IOException {
+    PowerMockito.doNothing().when(mNettyRemoteBlockWriter);
```
Comment (aaudiber) [inline core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java]: usually there's a method call after the `when`, what does this line do?

File: core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+import org.powermock.reflect.Whitebox;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+/**
+ * Unit tests for @{link NettyRemoteBlockWriter}.
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ClientContext.class, NettyRemoteBlockWriter.class})
```
Comment (aaudiber) [inline core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java]: does ClientContext.class need to be prepared?

**Comment by aaudiber** [comment]
Thanks for working on these tests @caadecarvalho, I've left some comments on the code. 

For each test please write the javadoc so that it says what functionality is being tested (e.g. `Tests that closing a file will update the writer's internal state to be closed`) vs what method is being tested (e.g. `Tests the close method`)

File: core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java (hunk 2 above)
Comment (caadecarvalho) [inline core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java]: OK, I'll do this.

File: core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java (hunk 3 above)
Comment (caadecarvalho) [inline core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java]: OK, I'll fix it.

File: core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java (hunk 5 above)
Comment (caadecarvalho) [inline core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java]: OK, I'll remove it.

File: core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java (hunk 4 above)
Comment (caadecarvalho) [inline core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java]: Yeah, it isn't fine, I'll try to make it work by creating a ServerBootstrap and correctly handle the RPCBlockWriteRequest. I'll let you know when I finish.

**Comment by caadecarvalho** [comment]
Great, @aaudiber . Sorry for the late answer, I'll let you know when I finish.

**Comment by gpang** [comment]
Hi @caadecarvalho , thanks for improving the codebase! Are there any updates on this PR?

**Comment by aaudiber** [comment]
@caadecarvalho Closing for now due to inactivity, feel free to reopen it if you have time!

**Pull request closed without merging**